app/src/main/java/com/itsaky/androidide/adapters/ProjectListAdapter.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.itsaky.androidide.R
+import com.itsaky.androidide.adapters.ProjectListAdapter.ViewHolder
+import com.itsaky.androidide.databinding.LayoutProjectListItemBinding
+import com.itsaky.androidide.models.ProjectInfoDetails
+
+class ProjectListAdapter(
+  projects: List<ProjectInfoDetails>,
+  private val onClick: ((ProjectInfoDetails, ViewHolder) -> Unit)? = null
+) : RecyclerView.Adapter<ViewHolder>() {
+
+  private val projects = projects.toMutableList()
+
+  class ViewHolder(internal val binding: LayoutProjectListItemBinding) :
+    RecyclerView.ViewHolder(binding.root)
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    return ViewHolder(LayoutProjectListItemBinding.inflate(
+      LayoutInflater.from(parent.context),
+      parent,
+      false
+    ))
+  }
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    holder.binding.apply {
+      val project = projects[position]
+      val context = root.context
+
+      val iconPath = project.file.absolutePath + "/app/src/main/res/mipmap-hdpi/ic_launcher.webp"
```

Comment:
It bad, hardcored icon path not good, maybe use a placeholder icon for all projects

Reply:
I've pondered this, and what's noteworthy is that if the icon isn't found, Glide seamlessly defaults to using the placeholder, which is a more favorable outcome. I believe that including the application icon itself would enhance the visualization of the application within the list.

Reply:
Another viable approach could involve searching for the application icon in AndroidManifest. However, it's important to note that the AndroidManifest structure might look something like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<adaptive icon xmlns:android="http://schemas.android.com/apk/res/android">
       <android background:drawable="@color/ic_launcher_background"/>
       <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
       <monochrome android:drawable="@mipmap/ic_launcher_foreground"/>
</adaptive-icon>
```

To achieve this, we use a LayerDrawable and combine the two components and we would have the application icon.